pkg/config/data.go

```diff
@@ -342,7 +347,18 @@ func (f *File) Load(endpoint, path string, c api.HTTPClient) error {
 		return fmt.Errorf("fetching remote configuration: expected '200 OK', received '%s'", resp.Status)
 	}
 
+	// NOTE: In an attempt to prevent unexpected changes to the in-memory data
```

Comment:
There are quite a few places throughout the CLI code where we call functions that ultimately end up mutating the in-memory data representation of the config (and in a couple of cases those calls happen asynchronously). 

Because the root cause of the bug causing profile data to be deleted is unclear, I wanted to take a 'belt and braces' approach and to synchronise any access to the in-memory data.